appengine-managed-runtime/src/main/java/com/google/apphosting/vmruntime/VmApiProxyDelegate.java

```diff
@@ -98,29 +97,46 @@
   static final int ADDITIONAL_HTTP_TIMEOUT_BUFFER_MS = 1000;
 
   protected int defaultTimeoutMs;
+
   protected final ExecutorService executor;
 
   protected final HttpClient httpclient;
 
-  final IdleConnectionMonitorThread monitorThread;
+  //final IdleConnectionMonitorThread monitorThread;
 
-  private static ClientConnectionManager createConnectionManager() {
-    PoolingClientConnectionManager connectionManager = new PoolingClientConnectionManager();
-    connectionManager.setMaxTotal(VmApiProxyEnvironment.MAX_CONCURRENT_API_CALLS);
-    connectionManager.setDefaultMaxPerRoute(VmApiProxyEnvironment.MAX_CONCURRENT_API_CALLS);
-    return connectionManager;
+  public VmApiProxyDelegate() {
+    this(createHttpClient());
   }
 
-  public VmApiProxyDelegate() {
-    this(new DefaultHttpClient(createConnectionManager()));
+  private static HttpClient createHttpClient() {
+    try {
+      HttpClientTransportOverHTTP transport = new HttpClientTransportOverHTTP();
+      // FIXME what sort of customize SslContextFactory we want to use here??
```

Comment:
It doesn't look like any requests go over SSL. So, we should be ok with not using one at all.